src/mlpack/methods/linear_svm/linear_svm_impl.hpp

```diff
@@ -66,6 +85,36 @@ LinearSVM<MatType>::LinearSVM(
   // No training to do here.
 }
 
+template <typename MatType>
+template <typename OptimizerType, typename... CallbackTypes>
+double LinearSVM<MatType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    OptimizerType optimizer,
+    CallbackTypes&&... callbacks)
+{
+  if (numClasses <= 1)
+  {
+    throw std::invalid_argument("LinearSVM dataset has 0 number of classes!");
+  }
+
+  LinearSVMFunction<MatType> svm(data, labels, numClasses, lambda, delta,
+      fitIntercept);
+  if (parameters.is_empty())
+    parameters = svm.InitialPoint();
+
+  // Train the model.
+  Timer::Start("linear_svm_optimization");
+  const double out = optimizer.Optimize(svm, parameters, callbacks...);
+  Timer::Stop("linear_svm_optimization");
+
+  Log::Info << "LinearSVM::LinearSVM(): final objective of "
+            << "trained model is " << out << "." << std::endl;
+
+  return out;
+}
```

Comment:
For a minute, I thought it would be nice to replace the implementation of the `Train()` overload that doesn't take callbacks with a call to this overload (with nothing as the callbacks).  But... I don't know how to call this overload from that one; I'm not sure it's possible.  So maybe the duplication can't be avoided.  If you know some way, let me know. :)